core/src/main/java/com/cloud/storage/template/HttpTemplateDownloader.java

```diff
@@ -218,7 +218,10 @@ public long download(boolean resume, DownloadCompleteCallback callback) {
             errorString = hte.getMessage();
         } catch (IOException ioe) {
             status = TemplateDownloader.Status.UNRECOVERABLE_ERROR; //probably a file write error?
-            errorString = ioe.getMessage();
+            // Let's not overwrite the original error message.
+            if (errorString.isEmpty()){
```

Comment:
errorString contains a single space initially. String.isEmpty returns (this.value.length == 0)
so this is never happening.

Reply:
You are correct @DaanHoogland, I have made the change.